app/gui/view/component-browser/component-list-panel/grid/src/entry/icon.rs

```diff
@@ -1,179 +1,55 @@
 //! All icons that are used in the Component Browser.
 
-
-// ==============
-// === Export ===
-// ==============
-
-pub use entry::Entry;
-pub use entry::Params;
-
-
-
-mod common_part;
-mod define_macro;
-mod entry;
```

Comment:
Big changes happened in this module. First of all, `common_part`, `define_macro`, and some common APIs were moved to a separate `icons` crate. Here we define only icons for the component entries. Section navigator icons are defined in the component-list-panel crate instead. `entry` module was also moved there.
This separation and refactoring are needed because we need to properly set up layer ordering for section navigator icons. They must be above the component list background, but this crate does not have a component list panel as a dependency. Hence the section navigator icons are defined in component-list-panel, and all the needed infrastructure for that is in a separate crate.

Reply:
I have another big issue with this, unfortunately. Icons should be defined in the same crate, because they cause performance issues now - every icon compiles as a separate shader, and thus, when we have even one instance of 50 icons, we are using 50 draw calls, which makes the app super slow. I'm planning to create a smart utility that merges all icons into one shader, but I need them to be defined in one crate (to be more precise, in one file). If you want to sort them correctly, just use more layers in such a case - will that work?

Reply:
Also, we want to register all icons in an "icon registry" that can be queried by icon names later. Basically, icons should be defined in one place. Also, @vitvakatu, remember that defining `above = [...]` is one of MANY ways to sort icons. We have also a dynamic sorting utility (see docs there) that you can use instead of this syntax and you can set it up from a code.